File: code/ts/modular_api/example/modules/time/time_builder.ts
import { ModuleBuilder } from '../../../src/index';
import { CurrentTimeInput, CurrentTimeOutput, CurrentTime } from './usecases/current_time';

export function buildTimeModule(m: ModuleBuilder): void {
  m.usecase('current-time', CurrentTime.fromJson, {
    inputClass: CurrentTimeInput,
    outputClass: CurrentTimeOutput,
    method: 'GET',
  });
}

File: code/dart/modular_api/example/modules/time/time_builder.dart
import 'package:modular_api/modular_api.dart';

import 'usecases/current_time.dart';

void buildTimeModule(ModuleBuilder m) {
  m.usecase(
    'current-time',
    CurrentTime.fromJson,
    inputExample: CurrentTimeInput.example,
    outputExample: CurrentTimeOutput.example,
    method: 'GET',
  );
}
